src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UpsertUnallocatedCaseService.kt

```diff
@@ -28,8 +28,9 @@ class UpsertUnallocatedCaseService(
     val storedUnallocatedEvents = repository.findByCrn(crn)
     workforceAllocationsToDeliusApiClient.getUserAccess(crn = crn)?.takeUnless { it.userExcluded || it.userRestricted }?.let {
       workforceAllocationsToDeliusApiClient.getUnallocatedEvents(crn)?.let { unallocatedEvents ->
-        log.debug("workforce to delius api client: getting unallocated events for crn: $crn")
+        log.debug("workforce to delius api client: getting unallocated events for crn $crn")
         val activeEvents = unallocatedEvents.activeEvents.associateBy { it.eventNumber.toInt() }
+        if (activeEvents.isEmpty()) { log.debug("No active events found for crn $crn") } else { log.debug("Active events found for crn $crn: $activeEvents") }
```

Comment:
so my understanding here is that `unallocatedEvents.activeEvents.associateBy` will give you a map of `AlloocatedEvent` objects. So the equivalent of this:

```
val = activeEvents = mapOf("1", ActiveEvent(
  eventNumber = "1",
  teamCode = "ZZZZ",
  providerCode = "PRO"
))
```
**Question**: will therefore logging the `activeEvents` map in the way you are above produce data in the log or the classic object reference log that will give your `@Objects5fe74` or something similar?

I guess you will get a stringified version of each `ActiveEvent` object (as it is a Kotlin data class and you get toString() for free) so that bit is fine. I'm just unsure if the Kotlin map with be stringified for free or you will end up with thre object reference in the log

Anyways, will approve as do not want to block and I guess can raise another PR if has this issue in the logs. Hopefully you get the map stringified for free!

If do not, can always debug this locally and add a watch of `activeEvents.toString()` to see what happens :)